code/crates/engine/src/consensus.rs

```diff
@@ -313,6 +274,9 @@ where
         state: &mut State<Ctx>,
         input: ConsensusInput<Ctx>,
     ) -> Result<(), ConsensusError<Ctx>> {
+        // Extract timeouts before the process! macro to avoid borrow checker issues
+        let timeouts = *state.consensus.timeouts();
```

Comment:
I think this is okay, but lmk

Reply:
I believe we can simplify this further by removing the timeouts from the core-consensus state, since it actually never uses them. See my attempt at https://github.com/melekes/malachite/pull/1, let me know what you think?